Javatrix.java

```diff
@@ -477,5 +477,31 @@ public Javatrix times(double scalar)
         }
         return actual;
     }
+
+    /**
+     * Gets the trace of a square matrix.
+     *
+     * @throws IllegalArgumentException if rows and
+     * columns are not the same.
+     * @return the trace of the matrix.
+     **/
+    public double trace() throws IllegalArgumentException
+    {
+        m = this.m;
+        n = this.n;
```

Comment:
should this be m = this.getRowDimension() for consistency?  Same for n and colDimension